[Bug] can't start broker  when enable acl2.0

### Before Creating the Bug Report

- [X] I found a bug, not just asking a question, which should be created in [GitHub Discussions](https://github.com/apache/rocketmq/discussions).

- [X] I have searched the [GitHub Issues](https://github.com/apache/rocketmq/issues) and [GitHub Discussions](https://github.com/apache/rocketmq/discussions)  of this repository and believe that this is not a duplicate.

- [X] I have confirmed that this bug belongs to the current repository, not other repositories of RocketMQ.


### Runtime platform environment

mac m1

### RocketMQ version

develop

### JDK Version

1.8

### Describe the Bug

broekr.conf

brokerClusterName = DefaultCluster
namesrvAddr = 127.0.0.1:9876
brokerName = broker-a
brokerId = 0
deleteWhen = 04
fileReservedTime = 48
storePathRootDir = /Users/zzl/01opensource/data
brokerRole = ASYNC_MASTER
flushDiskType = ASYNC_FLUSH
aclEnable = true
authenticationEnabled = true
authenticationProvider = org.apache.rocketmq.auth.authentication.provider.DefaultAuthenticationProvider
initAuthenticationUser = {"username":"rocketmq","password":"12345678"}
innerClientAuthenticationCredentials = {"accessKey":"rocketmq","secretKey":"12345678"}
authenticationMetadataProvider = org.apache.rocketmq.auth.authentication.provider.LocalAuthenticationMetadataProvider
<img width="839" alt="image" src="https://github.com/apache/rocketmq/assets/87265072/87631e95-0a52-4264-bff9-ba524d77c989">


### Steps to Reproduce

like before

### What Did You Expect to See?

start success

### What Did You See Instead?

org.apache.rocketmq.auth.authentication.exception.AuthenticationException: Init authentication user error.
	at org.apache.rocketmq.auth.authentication.manager.AuthenticationMetadataManagerImpl.initUser(AuthenticationMetadataManagerImpl.java:73)
	at org.apache.rocketmq.auth.authentication.manager.AuthenticationMetadataManagerImpl.<init>(AuthenticationMetadataManagerImpl.java:44)
	at org.apache.rocketmq.auth.authentication.factory.AuthenticationFactory.getMetadataManager(AuthenticationFactory.java:70)
	at org.apache.rocketmq.broker.BrokerController.<init>(BrokerController.java:348)
	at org.apache.rocketmq.broker.BrokerStartup.buildBrokerController(BrokerStartup.java:215)
	at org.apache.rocketmq.broker.BrokerStartup.createBrokerController(BrokerStartup.java:247)
	at org.apache.rocketmq.broker.BrokerStartup.main(BrokerStartup.java:52)
Caused by: java.util.concurrent.CompletionException: java.lang.IllegalStateException: The authenticationMetadataProvider is not configured
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292)
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:989)
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137)
	at org.apache.rocketmq.auth.authentication.manager.AuthenticationMetadataManagerImpl.initUser(AuthenticationMetadataManagerImpl.java:66)
	... 6 more
Caused by: java.lang.IllegalStateException: The authenticationMetadataProvider is not configured
	at org.apache.rocketmq.auth.authentication.manager.AuthenticationMetadataManagerImpl.getAuthenticationMetadataProvider(AuthenticationMetadataManagerImpl.java:211)
	at org.apache.rocketmq.auth.authentication.manager.AuthenticationMetadataManagerImpl.getUser(AuthenticationMetadataManagerImpl.java:164)
	... 7 more

### Additional Context

_No response_

--- a/auth/src/main/java/org/apache/rocketmq/auth/authentication/manager/AuthenticationMetadataManagerImpl.java
+++ b/auth/src/main/java/org/apache/rocketmq/auth/authentication/manager/AuthenticationMetadataManagerImpl.java
@@ -207,14 +207,14 @@ public class AuthenticationMetadataManagerImpl implements AuthenticationMetadata
     }
 
     private AuthenticationMetadataProvider getAuthenticationMetadataProvider() {
-        if (authorizationMetadataProvider == null) {
+        if (authenticationMetadataProvider == null) {
             throw new IllegalStateException("The authenticationMetadataProvider is not configured");
         }
         return authenticationMetadataProvider;
     }
 
     private AuthorizationMetadataProvider getAuthorizationMetadataProvider() {
-        if (authenticationMetadataProvider == null) {
+        if (authorizationMetadataProvider == null) {
             throw new IllegalStateException("The authorizationMetadataProvider is not configured");
         }
         return authorizationMetadataProvider;

